xwiki-platform-core/xwiki-platform-extension/xwiki-platform-extension-test/xwiki-platform-extension-test-pageobjects/src/main/java/org/xwiki/extension/test/po/ExtensionPane.java

```diff
@@ -187,10 +187,11 @@ private ExtensionPane clickAndWaitUntilElementIsVisible(WebElement button, Strin
      */
     private String getXPath()
     {
-        String nameAndVersion =
-            getDriver().findElementWithoutWaiting(container, By.className("extension-title")).getText();
+        String name = getDriver().findElementWithoutWaiting(container, By.className("extension-name")).getText();
+        String version = getDriver().findElementWithoutWaiting(container, By.className("extension-version")).getText();
         return String.format("//form[contains(@class, 'extension-item') and descendant::*["
-            + "contains(@class, 'extension-title') and normalize-space(.) = '%s']]", nameAndVersion);
+            + "contains(@class, 'extension-name') and normalize-space(.) = '%s'] and descendant::*["
+            + "contains(@class, 'extension-version') and normalize-space(.) = '%s']]", name, version);
```

Comment:
This is not good because it can match a **dependency** name or version. You need to limit the scope of the search, e.g. by looking only in the ``extension-header`` or ``extension-title``.

Reply:
Indeed. I just tried to fix this but XPath is new to me and I couldn't find the correct syntax so far. Can I ask you what would be a possible syntax for limiting the scope?

I tried this but there's something wrong with the `descendant` I guess and couldn't find much documentation yet.

```
return String.format("//form[contains(@class, 'extension-item') and descendant::*["
 + "*[@class = 'extension-header']//*[@class = 'extension-name'] and normalize-space(.) = '%s'] and "
 + "descendant::*[*[@class = 'extension-header']//*[@class = 'extension-version'] and normalize-space(.) = '%s']]",
```

Reply:
I have the feeling this complex xpath expression could be replaced by a simpler css selector. 
I haven't looking in more details at the code calling `getXPath` but that could be a way to solve your issue.